internal/httphandler/provenance_request_handler.go

```diff
@@ -149,7 +149,7 @@ func (p *provenanceRequestHandler) getDataReadByUser(w http.ResponseWriter, r *h
 	}
 	query := payload.(*types.GetDataReadByQuery)
 
-	response, err := p.db.GetValuesReadByUser(query.TargetUserId)
+	response, err := p.db.GetValuesReadByUser(query.UserId, query.TargetUserId)
 	if err != nil {
 		processInternalError(w, r, err)
```

Comment:
`processInternalError` must be extended (and probably renamed) to handle permission error correctly, right now it will return 500 where it should return 403

Reply:
Done